backend/models/time.py

```diff
@@ -0,0 +1,242 @@
+            source_points[0],
+            len(destination_points)
+        ))
+        request_url = 'https://api.mapbox.com/directions-matrix/v1/mapbox/driving/{points}'
+        all_points = source_points + destination_points
+        if len(all_points) > 25:
+            raise RuntimeError('Mapbox matrix API only accepts 25 locations.')
+        points = ';'.join([
+            '{longitude},{latitude}'.format(longitude=point[1], latitude=point[0])
+            for point in all_points
+        ])
+
+        params = {
+            'sources': ';'.join([str(i) for i, _ in enumerate(source_points)]),
+            'destinations': ';'.join(
+                [str(len(source_points) + i) for i, _ in enumerate(destination_points)]
+            ),
+            'access_token': self.api_key
+        }
+
+        response = requests.get(
+            url=request_url.format(points=points),
+            params=params
+        )
+
+        response.raise_for_status()
+        content = response.json()
+        return content['durations']
+
+    def measure_one_to_many_points(self, origin, point_list):
+        """
+        Measure duration from one to mamy points using the matrix API.
+
+        Result is a list of duration in  minutes.
+        """
+        measurements = []
+
+        for batch in iterators.iterate_in_slices(
+            iterable=iter(point_list),
+                batch_size=24):
+            measurements.extend(
+                self.get_matrix(
+                    source_points=[origin],
+                    destination_points=batch)[0]
+            )
+        return measurements
+
+    def closest_with_early_exit(self, origin, point_list, exit_distance):
+        """
+        Find closest point in a list of points, exiting early if exit_distance is reached.
+
+        The exit_distance uses haversine distance (in meters).
+        """
+        haversine_distances = [
+            self._haversine_measurer.measure_between_two_points(origin, point)
+            for point in point_list
+        ]
+        min_haversine_idx, min_haversine_distance = min(
+            enumerate(haversine_distances),
+            key=operator.itemgetter(1)
+        )
+
+        # TODO - Avoid early 'return' so that we can implement one call to
+        # `closest_with_early_exit` function building on `haversine` and `closest`,
+        # one return statement, and make it clearer as a result.
+        if min_haversine_distance <= exit_distance:
+            return (
+                min_haversine_distance / TIME_TO_DISTANCE_HOURS_TO_METERS * 60.0,
+                point_list[min_haversine_idx]
+            )
+
+        relevant_points = [
+            point_list[idx]
+            for idx, distance in enumerate(haversine_distances)
+            if distance < min(self.early_exit_outer_radius, 1.5 * min_haversine_distance)
+        ]
+
+        if not relevant_points:
+            return 10**42, point_list[0]
```

Comment:
Maybe make this a constant at the top of the file?

Reply:
Done.